server/data/data-cassandra/src/test/java/org/apache/james/user/cassandra/CassandraDelegationStoreTest.java

```diff
@@ -0,0 +1,85 @@
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.user.cassandra;
+
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+
+import java.util.Set;
+
+import org.apache.james.backends.cassandra.CassandraClusterExtension;
+import org.apache.james.core.Username;
+import org.apache.james.user.api.DelegationStore;
+import org.apache.james.user.api.DelegationStoreContract;
+import org.apache.james.user.api.UsersRepositoryException;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.junit.jupiter.api.extension.RegisterExtension;
+
+import com.github.fge.lambdas.Throwing;
+
+import reactor.core.publisher.Mono;
+
+public class CassandraDelegationStoreTest implements DelegationStoreContract {
+    @RegisterExtension
+    static CassandraClusterExtension cassandraCluster = new CassandraClusterExtension(CassandraUsersRepositoryModule.MODULE);
+
+    private final CassandraUsersDAO usersDAO = new CassandraUsersDAO(cassandraCluster.getCassandraCluster().getConf());
+
+    private CassandraDelegationStore testee;
+
+    @BeforeEach
+    void setUp() {
+        testee = new CassandraDelegationStore(usersDAO);
+        createBaseUsers(Set.of(ALICE, BOB, CEDRIC, DAMIEN));
+    }
+
+    @Override
+    public DelegationStore testee() {
+        return testee;
+    }
+
+    private void createBaseUsers(Set<Username> usernameSet) {
+        usernameSet.forEach(Throwing.consumer(username -> usersDAO.addUser(username, "secret")));
+    }
+
+    @Test
+    void addAuthorizedUserShouldThrowUsersRepositoryExceptionWhenBaseUserIsNotExisted() throws UsersRepositoryException {
+        usersDAO.removeUser(ALICE);
+
+        assertThatThrownBy(() -> Mono.from(testee.addAuthorizedUser(ALICE, BOB)).block())
+            .hasRootCause(new UsersRepositoryException(String.format("Unable to add delegate user. The base user %s is not existed.", ALICE.asString())));
+    }
+
+    @Test
+    void removeAuthorizedUserShouldThrowUsersRepositoryExceptionWhenBaseUserIsNotExisted() throws UsersRepositoryException {
+        usersDAO.removeUser(ALICE);
+
+        assertThatThrownBy(() -> Mono.from(testee.removeAuthorizedUser(ALICE, BOB)).block())
+            .hasRootCause(new UsersRepositoryException(String.format("Unable to remove delegate user. The base user %s is not existed.", ALICE.asString())));
+    }
+
+    @Test
+    void clearAlAuthorizedUserSShouldThrowUsersRepositoryExceptionWhenBaseUserIsNotExisted() throws UsersRepositoryException {
+        usersDAO.removeUser(ALICE);
+
+        assertThatThrownBy(() -> Mono.from(testee.clear(ALICE)).block())
+            .hasRootCause(new UsersRepositoryException(String.format("Unable to clear delegate users. The base user %s is not existed.", ALICE.asString())));
+    }
```

Comment:
IMO drop those tests.

We might want to use delegation on top of LDAP. With this constraint we cannot.